### DISCUSSION TITLE
disable lsp for certain file types

### DISCUSSION THREAD

**Discussion by zenlor** [helix-editor/helix#11624] [Q&A]
I have an annoying issue with LSP enabled for `*.hcl` files. By default `hcl` files are set to use `terraform-ls` but it usually is not right.
While it's true that a `.tf` has `hcl` syntax it's lsp setup is most likely wrong creating all sort of useless error reports.

Example:
<img width="807" alt="image" src="https://github.com/user-attachments/assets/a2197f0d-97c6-4554-9c33-28f954f6a452">

This is an `hcl` file, it doesn't contain terraform resources and it's not part of a terraform module, it's a terragrunt configuration file that orchestrates a terraform module.

Overriding the `hcl` definition works but I would have to re-write another entry for actual `tf` files using the same options as `hcl` except the lsp configuration.

Is there another way to _split_ file extensions in different language entries? Should I simply remove `.hcl` extension from the `hcl` filetype definition?

**Comment by zenlor** [comment]
Looks like I found a workaround. I'm modifying the `tfvars` language to include `tf` file extension while removing it from the `hcl` language:

```toml
[[language]]
name = "hcl"
file-types = ["hcl", "nomad"]
language-servers = []

[[language]]
name = "tfvars"
file-types = ["tf", "tfvars"]
```

The question still stands, is there a cleaner way to "inherit" (lacking a better word) a language definition overriding some of it's definitions?

**Reply by selvmaya** [reply]
I couldn't get it to work myself for my use case (`*.conf` interpreted as xml except without LSP), but [this discussion](https://github.com/helix-editor/helix/discussions/3109#discussioncomment-3188220) creates a language based on the (tree-sitter) grammar of another among other options.

See also https://github.com/helix-editor/helix/issues/3479#issuecomment-1221323100.
